src/server/repositories/interfaces/UrlRepositoryInterface.ts

```diff
@@ -0,0 +1,48 @@
+import { BaseRepository } from './BaseRepository'
+
+/**
+ * A url repository that handles access to the data store of Urls.
+ */
+export interface UrlRepositoryInterface extends BaseRepository<StorableUrl> {
+  findOne(where: object): Promise<StorableUrl | null>
```

Comment:
can we change the implementation to `findByShortUrl` that accepts a single string? since that is pretty much the only valid scenario in which you can `findOne` reliably.

Reply:
Then should we get rid of BaseRepository? Since its going to be different for each Repo